std/format.d

```diff
@@ -3692,30 +3706,27 @@ if (hasToString!(T, Char))
 {
     enum overload = hasToString!(T, Char);
 
-    static if (overload == 6)
+    static if (overload == HasToStringResult.customPutWriterFormatSpec)
     {
         val.toString(w, f);
     }
-    else static if (overload == 5)
+    else static if (overload == HasToStringResult.customPutWriter)
     {
         val.toString(w);
     }
-    // not using the overload enum to not break badly defined toString overloads
-    // e.g. defining the FormatSpec as ref and not const ref led this function
-    // to ignore that toString overload
```

Comment:
@JackStouffer the `constCharSinkFormatSpec` overload is:

```
T val = void; val.toString((const(char)[] s){}, "%s");
```

Reply:
that's the `constCharSinkFormatString` overload